lib/spack/spack/test/concretize_preferences.py

```diff
@@ -270,7 +270,9 @@ def test_external_mpi(self):
         # ensure that once config is in place, external is used
         spec = Spec("mpi")
         spec.concretize()
-        assert spec["mpich"].external_path == os.sep + os.path.join("dummy", "path")
+        # This is a hardcoded value from the conf above so correct
+        # platform specific paths do not need to be used here
+        assert spec["mpich"].external_path == os.path.sep + os.path.join("dummy", "path")
```

Comment:
This check is confusing WRT the comment that was added:

* `os.path.sep` == `os.sep` AFAIK (so this test hasn't changed)
* The comment says platform specific paths do not need to be accounted for, but it looks like that's what this is doing by using `os.path.join` (if it were fine to ignore, I assume it would be using something like `posixpath`).

Reply:
Oh I meant to remove the comment. That was from a previous (and incorrect) fix, that did fix this test, but did not fix the underlying problem.